File: 2419/Solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestSubarray = function(nums) {
    let max = 0;
    let cur = 0;
    let res = 0;
    for(let i of nums){
        max = Math.max(max, i);
    }
    for(let i of nums){
        if(i == max){
            cur++;
            res = Math.max(res, cur);
        }else{
            cur = 0;
        }
    }
    return res;
};

File: 2419/Solution.java
class Solution {
    public int longestSubarray(int[] nums) {
        int max = 0, cur = 0, res = 0;
        for(int i : nums) max = Math.max(max, i);
        for(int i : nums){
            if(max == i){
                cur++;
                res = Math.max(res, cur);
            }else{
                cur = 0;
            }
        }
        return res;
    }
}
